build.gradle

```diff
@@ -11,23 +11,39 @@ buildscript {
     }
 }
 
+ext {
+    groovyVersion = System.getenv('CI_GROOVY_VERSION') ?: project.groovyVersion
+    isCiBuild = System.getenv().get("TRAVIS") == 'true' || (System.getenv().get("CI") as Boolean)
+    isSnapshot = project.projectVersion.endsWith('-SNAPSHOT')
+    isReleaseVersion = !isSnapshot
+}
+
 repositories {
     mavenCentral()
     maven { url "https://repo.grails.org/grails/core" }
+    if(groovyVersion.endsWith('-SNAPSHOT')) {
+        maven {
+            name = 'ASF Snapshot repo'
+            url = 'https://repository.apache.org/content/repositories/snapshots'
+        }
+    }
 }
 
 group = "org.grails"
 version project.projectVersion
 
-ext {
-    groovyVersion = System.getenv('CI_GROOVY_VERSION') ?: project.groovyVersion
-    isCiBuild = System.getenv().get("TRAVIS") == 'true' || (System.getenv().get("CI") as Boolean)
-    commonBuild = 'https://raw.githubusercontent.com/grails/grails-common-build/v2.0.3'
-}
+apply plugin: 'java-library'
+apply plugin: 'groovy'
+
+ext."signing.keyId" = project.hasProperty("signing.keyId") ?: System.getenv('SIGNING_KEY')
+ext."signing.secretKeyRingFile" = project.hasProperty("signing.secretKeyRingFile") ?: "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg"
+ext."signing.password" = project.hasProperty("signing.password") ?: System.getenv('SIGNING_PASSPHRASE')
```

Comment:
Actually, this is not going to work as intended.
It should be using `project.findProperty('x')`:
```groovy
ext.set('signing.keyId', project.findProperty('signing.keyId') ?: System.getenv('SIGNING_KEY'))
ext.set('signing.secretKeyRingFile', project.findProperty('signing.secretKeyRingFile') ?: "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg")
ext.set('signing.password', project.findProperty('signing.password') ?: System.getenv('SIGNING_PASSPHRASE'))
```